### PR TITLE
docs(recipes/necord): add necord documentation

### PR DESCRIPTION
**Pull request by SocketSomeone**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
To introduce new library [necord](https://github.com/necordjs/necord)

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

## Other information
Already added to https://github.com/nestjs/awesome-nestjs and Discord Server as Forum channel

### CODE REVIEW

File: content/recipes/necord.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    @UseInterceptors(AnimeAutocompleteInterceptor)
+    @SlashCommand({
+        name: 'anime',
+        description: 'Lookup information about an anime'
+    })
+    public async onSearch(@Context() [interaction]: SlashCommandContext, @Options() { anime }: AnimeDto) {
+        return interaction.reply({content: `I found the anime ${anime}`});
+    }
+}
+```
```
Comment (jmcdo29) [inline content/recipes/necord.md:247]: Personally, I love this! It's awesome! But to keep it more inline with the rest of our docs, do you think you could make it a `CatsAutocompleteInterceptor` that checks for cat breeds and adjust the command and descriptions accordingly?

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
Overall, the docs look fantastic. Could you revert the whitespace changes in the menu.components and recipes.module? I don't think converting the whitespace to tabs was intentional, just a side effect of a save command of your IDE. 

@kamilmysliwiec I can vouch for this library, I've used it for some trivial Discord integration bots before and will keep using it and probably make a Nest bot for our server as well when I've got some time.

File: content/recipes/necord.md (hunk 1 above)
Comment (SocketSomeone) [reply content/recipes/necord.md:247]: @jmcdo29 Yes! Of course, I will make changes within a few hours, already reverted the spaces

Comment (SocketSomeone) [reply content/recipes/necord.md]: @jmcdo29 Updated autocomplete category at [13f2ea3](https://github.com/nestjs/docs.nestjs.com/pull/2656/commits/13f2ea379a501b9e7540f12913ec87f008f7c05f)

**Review by jmcdo29 — APPROVED** [review]
LGTM. We'll see what Kamil thinks

File: content/recipes/necord.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+All of built-in select menu components:
+- `@StringSelect`
+- `@UserSelect`
+- `@RoleSelect`
+- `@MentionableSelect`
+- `@ChannelSelect`
```
Comment (micalevisk) [inline content/recipes/necord.md:341]: I wonder if we being too descriptive here is needed. If the package change over time, we would have to update here as well... Instead, we could replace this with a link to the package's docs and only show general capabilities of the package

**Review by micalevisk — APPROVED** [review]

**Comment by KieronWiltshire** [comment]
Any update on this please?

**Comment by kamilmysliwiec** [comment]
While I appreciate your effort and I'm excited to see more and more great OSS Nest integrations, I'm not convinced that we should keep adding them to the Recipes section. This category (Recipes) already contains several packages that are not maintained by us (are not part of the official NestJS organization) and so the documentation got quite difficult to maintain moving forward as we're not familiar with the API of those packages ourselves

**Comment by KieronWiltshire** [comment]
> While I appreciate your effort and I'm excited to see more and more great OSS Nest integrations, I'm not convinced that we should keep adding them to the Recipes section. This category (Recipes) already contains several packages that are not maintained by us (are not part of the official NestJS organization) and so the documentation got quite difficult to maintain moving forward as we're not familiar with the API of those packages ourselves

while I agree with you somewhat, I would argue that NestJS is the go-to framework for building discord bots. Necord is hitting 8.3k downloads per month.

**Comment by SocketSomeone** [comment]
@kamilmysliwiec

Hi there,
Thank you for answer!

> that are not maintained by us (are not part of the official NestJS organization) and so the documentation got quite difficult to maintain moving forward as we're not familiar with the API of those packages ourselves

I understand your concerns, that's why we are making the information more generalized, adding links to our own documentation for advanced info and a hint that library not managed by NestJS. It will also not be difficult for me to maintain documentation on your site, and there are also many cool guys who are ready to help with this work. I am constantly monitoring any changes, including the necord channel (Discord server).

> Necord is hitting 8.3k downloads per month.

It's true that the NECORD community is growing now and we want as many people as possible to see it. As @jmcdo29 said, he also wants to create a bot for your server using NECORD.

> This category (Recipes) already contains several packages

I also agree with @samchon (https://github.com/nestjs/docs.nestjs.com/pull/2571#issuecomment-1495913962). It would be cool to create a new category for great OSS Nest integrations not managed by Nest.

> While I appreciate your effort and I'm excited to see more and more great OSS Nest integrations

We love you guys no matter what! NECORD 💖 NestJS
Thanks for your hard work!

**Comment by kamilmysliwiec** [comment]
I'm still hesitant about adding yet another recipe to the docs but if @jmcdo29 and @micalevisk say we should do this then let's move ahead.

Thanks @SocketSomeone, fantastic job!

**Comment by smolinari** [comment]
@kamilmysliwiec 

How about allowing the community to create a new site like "Built for Nest", where it could become a community maintained reference resource of community shared Nest modules? This could help keep the Nest docs Nest concentrated and yet give the community a more centralized place to offer their work and documentation to the rest of the Nest community? It would be THE source of truth for community shared Nest modules and documentation, which in turn would also help promote Nest in the end. We'd just need some ground rules as to what the content can be and a couple of curators/ repo maintainers. I'd be willing to help. 😁 

Scott

**Comment by xKrvZ** [comment]
> @kamilmysliwiec
> 
> How about allowing the community to create a new site like "Built for Nest", where it could become a community maintained reference resource of community shared Nest modules? This could help keep the Nest docs Nest concentrated and yet give the community a more centralized place to offer their work and documentation to the rest of the Nest community? It would be THE source of truth for community shared Nest modules and documentation, which in turn would also help promote Nest in the end. We'd just need some ground rules as to what the content can be and a couple of curators/ repo maintainers. I'd be willing to help. 😁
> 
> Scott

There is nothing stopping you from creating this site. If it's community driven, I wouldn't see why you'd need support for this.

**Comment by smolinari** [comment]
> > @kamilmysliwiec
> 
> There is nothing stopping you from creating this site. If it's community driven, I wouldn't see why you'd need support for this.

True. But it wouldn't be an official Nest site either. That's what I'm getting at. 

Scott

**Comment by kamilmysliwiec** [comment]
@smolinari agree that currently cherry-picked recipes, favoring certain packages vs another feels unfair. im gonna go ahead and remove some of them, then we could figure something out for a community-driven page(?). Not sure how/what yet

**Pull request merged**